src/Samples/dotnet6server/Program.cs

```diff
@@ -1,24 +1,42 @@
 using CoreWCF;
 using CoreWCF.Configuration;
+using CoreWCF.Channels;
+using CoreWCF.Description;
 using dotnet6Server;
 using MyContracts;
 
+const string BASE_HTTP_URL = "http://localhost:5000";
+const string BASE_HTTPS_URL = "https://localhost:5001";
+
 var builder = WebApplication.CreateBuilder(args);
 builder.WebHost.ConfigureKestrel((context, options) =>
 {
     options.AllowSynchronousIO = true;
 });
 
-builder.Services.AddServiceModelServices();
+// Add WSDL support
+builder.Services.AddServiceModelServices().AddServiceModelMetadata();
+builder.Services.AddSingleton<IServiceBehavior, UseRequestHeadersForMetadataAddressBehavior>();
 var app = builder.Build();
 
+app.Urls.Add(BASE_HTTP_URL);
+app.Urls.Add(BASE_HTTPS_URL);
+
 app.UseServiceModel(builder =>
 {
-    builder.AddService<EchoService>()
-    .AddServiceEndpoint<EchoService, IEchoService>(new BasicHttpBinding(), "/EchoService/basichttp");
+    builder.AddService<EchoService>((serviceOptions) =>
+    {
+        // Set the base addressrs that will be used for the service and acts as the WSDL endpoint 
+        serviceOptions.BaseAddresses.Add(new Uri($"{BASE_HTTP_URL}/EchoService"));
```

Comment:
@mconnew - The way that asp.net core works means that if you configure URLs via config or the cmd line, you don't know what base will be. That's why I added them explicitly. It would be better if the base address could be set using a relative path to the host, and not need to know the full uri here.

Reply:
@samsp-msft i think that you can simply let the runtime get the urls from `launchsettings.json` and/or env variable or `--urls` argument. Please see https://github.com/g7ed6e/CoreWCF/blob/feature/project-template/src/CoreWCF.Templates/src/templates/CoreWCFService/Program.cs